src/maps.rs

```diff
@@ -141,10 +150,40 @@ where
 }
 
 /// Parse the maps file for the process with the given PID.
-pub(crate) fn parse(pid: Pid) -> Result<Vec<MapsEntry>, Error> {
+pub(crate) fn parse<F>(pid: Pid, filter: F) -> Result<Vec<MapsEntry>, Error>
+where
+    F: FnMut(&MapsEntry) -> bool,
+{
     let path = format!("/proc/{pid}/maps");
     let file = File::open(path)?;
-    parse_file(file, pid)
+    parse_file(file, pid, filter)
+}
+
+/// A helper function checking whether a `MapsEntry` has relevant to
+/// symbolization efforts. If that is not the case, it may be possible to ignore
+/// it altogether.
+pub(crate) fn is_symbolization_relevant(entry: &MapsEntry) -> bool {
+    // Only entries with actual paths are of relevance.
+    if entry.path.as_path().components().next() != Some(Component::RootDir) {
+        return false
+    }
+
+    // Only entries that are executable and readable (r-x-) are of relevance.
+    if (entry.mode & 0b1010) != 0b1010 {
+        return false
+    }
+
+    if let Ok(meta_data) = entry.path.metadata() {
+        if !meta_data.is_file() {
```

Comment:
is this doing `stat()` syscall for each entry? If yes, can we avoid that?

Reply:
> is this doing `stat()` syscall for each entry?

I think so.

> If yes, can we avoid that?

We should be able to minimize it with some caching scheme, but I am not entirely sure why this check is really needed. Anyway, this change just moves code around.

Reply:
yeah, if we can get rid of unnecessary checks like this, it probably would be best, thanks!

Reply:
Will remove it subsequently.